scripts/lib/CIME/XML/component.py

```diff
@@ -282,3 +282,46 @@ def print_values(self):
             name = self.get(entry, "id")
             text = self.text(self.get_child("desc", root=entry))
             logger.info("   {:20s} : {}".format(name, text.encode('utf-8')))
+
+
+    def return_values(self):
+        """
+        return a list of hashes from target config_component.xml file
+        """
+        entry_dict = dict()
+        items = list()
+        try:
+            helpnode = self.get_child("help")
+            helptext = self.text(helpnode)
+        except:
+            helptext = ''
+            
+        entries = self.get_children("entry")
+        for entry in entries:
+            item = dict()
+            name = self.get(entry, "id")
+            datatype = self.text(self.get_child("type", root=entry))
+            try:
+                valid_values = self.text(self.get_child("valid_values", root=entry))
+            except:
+                valid_values = ''
+
+            try:
+                value = self.get_value(name)
+            except:
+                value = ''
+            
```

Comment:
"try and except" is the wrong construct here,  I think you want to check if node valid_values exists and only call self.text if it does.  If it exists and has no text valid_values will be ''.   Same with the other one.